### PR TITLE
Fixed gitlab_*_variable when using purge

### LINKED ISSUE #7250: gitlab_*_variables deletes all variables in purge mode
### Summary

When using community.general 7.4 to provision gitlab with an exact list of variables (purge:true) all variables get deleted. This took our GitLab env with more then 1000 repos fully offline ;-)

This is due to the change done to the modules without adding the new 'raw' parameter to the list of known parameters in the module_utils/gitlab

### Issue Type

Bug Report

### Component Name

gitlab_group_variable
gitlab_project_variable

### Ansible Version

does not matter

### Community.general Version

7.4

### Configuration

_No response_

### OS / Environment

any

### Steps to Reproduce

```yaml (paste below)
---
- hosts: localhost
  gather_facts: false
  vars:
    gv:
      Test:
        value: "12345"
        masked: false
        protected: false
        variable_type: env_var

  tasks:
    - name: test
      community.general.gitlab_group_variable:
        api_url: "https://gitlab/"
        api_token: "secrettoken
        group: "an existing gitlab group"
        purge: true
        vars: "{{ gv }}"
```

### Expected Results

The group varibale Test is set, and no other variable

### Actual Results

no variables are set at all

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by MaxBidlingmaier**
##### SUMMARY
Hotfix to 7.4 release deleting all GitLab variables

Fixes #7250 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME

affected modules are:
- gitlab_group_variable
- gitlab_project_variable

##### ADDITIONAL INFORMATION
```paste below

```

### CODE REVIEW

File: plugins/modules/gitlab_group_variable.py [hunk 1]
```diff
@@ -381,6 +381,8 @@ def main():
         group=dict(type='str', required=True),
         purge=dict(type='bool', required=False, default=False),
         vars=dict(type='dict', required=False, default=dict(), no_log=True),
+        # please mind whenever chaning the variables dict to also change module_utils/gitlab.py's
```
Comment (lgatellier) [inline plugins/modules/gitlab_group_variable.py:384]:
```suggestion
        # please mind whenever changing the variables dict to also change module_utils/gitlab.py's
```

File: plugins/modules/gitlab_project_variable.py [hunk 2]
```diff
@@ -404,6 +404,8 @@ def main():
         project=dict(type='str', required=True),
         purge=dict(type='bool', required=False, default=False),
         vars=dict(type='dict', required=False, default=dict(), no_log=True),
+        # please mind whenever chaning the variables dict to also change module_utils/gitlab.py's
```
Comment (lgatellier) [inline plugins/modules/gitlab_project_variable.py:407]:
```suggestion
        # please mind whenever changing the variables dict to also change module_utils/gitlab.py's
```

File: changelogs/fragments/7251-gitlab-variables-deleteing-all-variables.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - "gitlab_project_variable deletes all variables when used with purge: true due to missing 'raw' property in KNOWN attributes"
+  - "gitlab_group_variable deletes all variables when used with purge: true due to missing 'raw' property in KNOWN attributes"
```
Comment (lgatellier) [inline changelogs/fragments/7251-gitlab-variables-deleteing-all-variables.yml:3]:
```suggestion
  - gitlab_project_variable deletes all variables when used with ``purge=true`` due to missing ``raw`` property in KNOWN attributes (https://github.com/ansible-collections/community.general/issues/7250).
  - gitlab_group_variable deletes all variables when used with ``purge=true`` due to missing ``raw`` property in KNOWN attributes (https://github.com/ansible-collections/community.general/issues/7250).
```

**Review by lgatellier — CHANGES_REQUESTED** [review]
Thanks for your contribution !

Can you squash all your commits in a single self-explaining commit ?

Would you also mind adding an integration test to ensure this bug won't occur in eventual future variable attributes change ?

File: changelogs/fragments/7251-gitlab-variables-deleteing-all-variables.yml (hunk 3 above)
Comment (felixfontein) [reply changelogs/fragments/7251-gitlab-variables-deleteing-all-variables.yml:3]:
```suggestion
  - "gitlab_project_variable - deleted all variables when used with ``purge: true`` due to missing ``raw`` property in KNOWN attributes (https://github.com/ansible-collections/community.general/issues/7250, https://github.com/ansible-collections/community.general/pull/7251)."
  - "gitlab_group_variable - deleted all variables when used with ``purge: true`` due to missing ``raw`` property in KNOWN attributes (https://github.com/ansible-collections/community.general/issues/7250, https://github.com/ansible-collections/community.general/pull/7251)."
```

**Comment by MaxBidlingmaier** [comment]
> Thanks for your contribution !
> 
> Can you squash all your commits in a single self-explaining commit ?
> 
> Would you also mind adding an integration test to ensure this bug won't occur in eventual future variable attributes change ?

Hm, I think the whole code of the two modules somehow got  broken since I contributed some time ago, so it would be better to fix the code then work around the problem with additional tests. 
For purging we do not need to compare the full property list (where depending on the GitLab version the structure could be different). In fact the current code only works for spefcific GitLab versions. So if we would change to just compare variable names again for that part we would ensure that it cannot happen.

File: changelogs/fragments/7251-gitlab-variables-deleteing-all-variables.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - gitlab_project_variable deletes all variables when used with ``purge=true`` due to missing ``raw`` property in KNOWN attributes (https://github.com/ansible-collections/community.general/issues/7250).
+  - gitlab_group_variable deletes all variables when used with ``purge=true`` due to missing ``raw`` property in KNOWN attributes (https://github.com/ansible-collections/community.general/issues/7250).
```
Comment (felixfontein) [inline changelogs/fragments/7251-gitlab-variables-deleteing-all-variables.yml:3]:
```suggestion
  - gitlab_project_variable - deleted all variables when used with ``purge=true`` due to missing ``raw`` property in KNOWN attributes (https://github.com/ansible-collections/community.general/issues/7250).
  - gitlab_group_variable - deleted all variables when used with ``purge=true`` due to missing ``raw`` property in KNOWN attributes (https://github.com/ansible-collections/community.general/issues/7250).
```

**Comment by felixfontein** [comment]
@lgatellier is it OK for you to merge this, so that the issue is fixed? I guess additional things (more tests, ...) can still be added in follow-up PRs.

**Comment by lgatellier** [comment]
> @lgatellier is it OK for you to merge this, so that the issue is fixed? I guess additional things (more tests, ...) can still be added in follow-up PRs.

Yep, LGTM ! 🚀

**Comment by felixfontein** [comment]
@MaxBidlingmaier thanks for fixing this!
@lgatellier thanks for reviewing!

**Comment by markuman** [comment]
Oh damn, it was introduced while adding the `raw` feature https://github.com/ansible-collections/community.general/pull/7132  
thx for fixing!

**Comment by markuman** [comment]
I wonder if the integration test covers it ...

**Comment by markuman** [comment]
yeah, integration test would have covered it :-/

```
TASK [Install required libs] *****************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [purge all variables for check_mode test] ***********************************************************************************************************************************************************************************************
changed: [localhost]

TASK [add a variable value in check_mode] ****************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [check_mode state must be changed] ******************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [apply add value from check_mode test] **************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [state must be changed] *****************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [test new format] ***********************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [state must be not changed] *************************************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {
    "assertion": "gitlab_project_variable_state is not changed",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
localhost                  : ok=8    changed=4    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

```

So we lack here on CI against a gitlab instance.  
Maybe someone has some time to create an integration test against a gitlab container in github actions.

**Pull request merged**